### PR TITLE
Remove unnecessary ledger rpcs

### LINKED ISSUE #866: Remove Ledger RPC methods we don't need
method list to be created.

to be removed:

- [x] ledger_getHead
- [x] ledger_getSlots
- [x] ledger_getBatches
- [x] ledger_getTransactions
- [x] ledger_getEvents
- [x] ledger_getSlotByHash
- [x] ledger_getBatchByHash
- [x] ledger_getTransactionByHash
- [x] ledger_getSlotByNumber
- [x] ledger_getBatchByNumber
- [x] ledger_getTransactionByNumber
- [x] ledger_getEventByNumber
- [x] ledger_getSlotsRange
- [x] ledger_getBatchesRange
- [x] ledger_getTransactionsRange
- [x] ledger_subscribeSlots

anything with "SoftBatch" in its name should be renamed to "SoftConfirmation"

### PR DESCRIPTION
**Pull request by ercecan**
# Description
Removes unnecessary ledger rpcs

## Linked Issues
- Fixes #866

### CODE REVIEW

File: crates/fullnode/tests/runner_reorg_tests.rs [hunk 1]
```diff
@@ -1,217 +0,0 @@
...
-use sov_rollup_interface::services::da::DaService;
-use sov_rollup_interface::storage::HierarchicalStorageManager;
-use sov_state::storage::NativeStorage;
-use sov_state::{ProverStorage, Storage};
-
-type MockInitVariant =
-    InitVariant<HashStf<MockValidityCond>, MockZkvm<MockValidityCond>, MockDaSpec>;
-#[tokio::test]
-#[ignore]
-async fn test_simple_reorg_case() {
```
Comment (kpp) [inline crates/fullnode/tests/runner_reorg_tests.rs:27]: Maybe we should keep them?

File: crates/sovereign-sdk/full-node/sov-ledger-rpc/src/server.rs [hunk 2]
```diff
@@ -298,11 +157,9 @@ where
     Ok(rpc)
 }
 
-#[derive(serde::Deserialize)]
-struct RangeArgs(u64, u64, #[serde(default)] QueryMode);
-
 /// A structure containing serialized query arguments for RPC queries.
 #[derive(serde::Deserialize)]
+#[allow(dead_code)]
```
Comment (kpp) [inline crates/sovereign-sdk/full-node/sov-ledger-rpc/src/server.rs:162]: No

File: crates/fullnode/tests/runner_reorg_tests.rs (hunk 1 above)
Comment (ercecan) [reply crates/fullnode/tests/runner_reorg_tests.rs:27]: Yeah maybe, @eyusufatik  What do you think?

**Review by eyusufatik — CHANGES_REQUESTED** [review]
delete ledger db tables please

**Comment by eyusufatik** [comment]
> we are good to go if you delete (now) unused ledger db tables

left review so that it would be easier to review later on

File: bin/citrea/tests/e2e/mod.rs [hunk 3]
```diff
@@ -403,7 +403,7 @@ async fn test_all_flow() {
 
 /// Test RPC `ledger_getHeadSoftBatch`
```
Comment (kpp) [inline bin/citrea/tests/e2e/mod.rs:404]: fix comment

File: bin/citrea/src/test_rpc.rs
Comment (eyusufatik) [inline bin/citrea/src/test_rpc.rs:1]: create issue for our ledger rpc tests

File: crates/sovereign-sdk/full-node/sov-ledger-rpc/tests/empty_ledger.rs
Comment (eyusufatik) [inline crates/sovereign-sdk/full-node/sov-ledger-rpc/tests/empty_ledger.rs:1]: write same tests for our functions maybe?

File: crates/sovereign-sdk/full-node/db/sov-db/src/schema/tables.rs [hunk 4]
```diff
@@ -53,16 +53,15 @@ pub const STATE_TABLES: &[&str] = &[
 pub const LEDGER_TABLES: &[&str] = &[
     SlotByNumber::table_name(),
```
Comment (eyusufatik) [inline crates/sovereign-sdk/full-node/db/sov-db/src/schema/tables.rs:54]: we don't use slotbynumber by hash do tx by hash by number, event by ...

Comment (eyusufatik) [reply crates/sovereign-sdk/full-node/db/sov-db/src/schema/tables.rs:54]: did we not remove these because it was breaking too many tests?

Comment (ercecan) [reply crates/sovereign-sdk/full-node/db/sov-db/src/schema/tables.rs:54]: We use them only for setting next item numbers

File: bin/citrea/src/test_rpc.rs
Comment (ercecan) [reply bin/citrea/src/test_rpc.rs:1]: #939

File: crates/sovereign-sdk/full-node/db/sov-db/src/schema/tables.rs (hunk 4 above)
Comment (ercecan) [reply crates/sovereign-sdk/full-node/db/sov-db/src/schema/tables.rs:56]: Talked with @eyusufatik and decided removing tables should be a separate pr

**Review by eyusufatik — APPROVED** [review]

**Pull request merged**